web/vue.config.js

```diff
@@ -1,6 +1,15 @@
 const MomentLocalesPlugin = require('moment-locales-webpack-plugin');
+const child_process = require('child_process');
 
-process.env.VUE_APP_VERSION = process.env.COMMIT_REF;
+function getVersion() {
+  try {
+    return child_process.execSync("git describe").toString();
+  } catch (err) {
+    return process.env.COMMIT_REF;
+  }
+}
+
+process.env.VUE_APP_VERSION = getVersion();
```

Comment:
If `git describe` and `COMMIT_REF` both fail to produce a value, than the value of `process.env.VUE_APP_VERSION` will be `'undefined'` (a string), since [any values assigned to `process.env` are coerced as such](https://nodejs.org/dist/latest-v16.x/docs/api/process.html#process_process_env).

Maybe that string is exactly what you'd like to render on the footer component, but since this is implicit, I wanted to point it out.